osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs

```diff
@@ -60,11 +60,23 @@ public override void Add(DrawableHitObject h)
 
             var c = h as IDrawableHitObjectWithProxiedApproach;
             if (c != null)
-                approachCircles.Add(c.ProxiedLayer.CreateProxy());
+            {
+                var original = c.ProxiedLayer;
+                // lifetime is set on LoadComplete so wait until it.
+                original.OnLoadComplete += addApproachCircleProxy;
```

Comment:
I want to see `ProxyDrawable` handle this stuff internally, perhaps by binding to the original drawable's `LifetimeChanged` events and copying the lifetime.

Additionally I think this will fail in the editor when we move hitobjects in time (hasn't been implemented yet).